### PR TITLE
chore: trim the query when searching for wallets

### PR DESCRIPTION
**Pull request by lukaisailovic**
When searching for wallets we would send any whitespace as well which would present us with "no wallets" screen. 

Made two changes:
- Sending `trimmedQuery` to ApiController instead of `query`
- Compare `trimmedQuery` with previous query that is also trimmed. This will not send network calls for whitespaces

### CODE REVIEW

**Comment by lukaisailovic** [comment]
> Wondering if this should be inside of ApiController directly + test? 

@tomiir Thats a good point. I think I will have to trim on the UI layer to not trigger additional requests (changed if condition).

Do you think we should add trimming to the controller in addition to the current logic?

**Comment by chris13524** [comment]
+1 on a test for this

File: packages/core/tests/controllers/ApiController.test.ts [hunk 1]
```diff
@@ -383,6 +383,64 @@ describe('ApiController', () => {
...
+      }
+    ]
+    OptionsController.setIncludeWalletIds(includeWalletIds)
+    OptionsController.setExcludeWalletIds(excludeWalletIds)
+
+    let fetchSpy = vi.spyOn(api, 'get').mockResolvedValue({ data })
+    const fetchImageSpy = vi.spyOn(ApiController, '_fetchWalletImage').mockResolvedValue()
+
+    // Whitespace
+    await ApiController.searchWallet({ search: 'MetaMask    ' })
```
Comment (chris13524) [inline packages/core/tests/controllers/ApiController.test.ts:404]: Test with leading whitespace not just trailing would be helpful too

Comment (lukaisailovic) [reply packages/core/tests/controllers/ApiController.test.ts:404]: @chris13524 Done!

**Review by tomiir — APPROVED** [review]

**Pull request merged**